examples/completion/main.go

```diff
@@ -0,0 +1,96 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+func listHosts(args []string) []string {
+	// Provide a dynamic list of hosts from a hosts file or otherwise
+	// for bash completion. In this example we simply return static slice.
+
+	// You could use this functionality to reach into a hosts file to provide
+	// completion for a list of known hosts.
+	return []string{"sshhost.example", "webhost.example", "ftphost.example"}
+}
+
+type NetcatCommand struct {
+	hostName string
+	port     int
+	format   string
+}
+
+func (n *NetcatCommand) run(c *kingpin.ParseContext) error {
+	fmt.Printf("Would have run netcat to hostname %v, port %d, and output format %v\n", n.hostName, n.port, n.format)
+	return nil
+}
+
+func configureNetcatCommand(app *kingpin.Application) {
+	c := &NetcatCommand{}
+	nc := app.Command("nc", "Connect to a Host").Action(c.run)
+	nc.Flag("nop-flag", "Example of a flag with no options").Bool()
+
+	// You can provide hint options using a function to generate them
+	nc.Flag("host", "Provide a hostname to nc").
+		Required().
+		HintAction(listHosts).
+		StringVar(&c.hostName)
+
+	// You can provide hint options statically
+	nc.Flag("port", "Provide a port to connect to").
+		Required().
+		HintOptions("80", "443", "8080").
```

Comment:
This is a good example of where `HintOptions()` could be useful. Maybe put that in the README as well?